test/functional/tests/io_class/test_io_class_purge.py

```diff
@@ -0,0 +1,148 @@
+        description: |
+          Create ioclasses for 3 different directories. Run IO against each
+          directory, check usage stats correctness before and after purge
+        pass_criteria:
+          - Usage stats are consistent on each test step
+          - Usage stats doesn't exceed cache size
+    """
+    with TestRun.step("Prepare disks"):
+        cache, core = prepare()
+        cache_size = cache.get_statistics().config_stats.cache_size
+
+    with TestRun.step("Disable udev"):
+        Udev.disable()
+
+    with TestRun.step(f"Preparing filesystem and mounting {core.system_path} at {mountpoint}"):
+        filesystem = Filesystem.xfs
+        core.create_filesystem(filesystem)
+        core.mount(mountpoint)
+        sync()
+
+    with TestRun.step("Prepare test dirs"):
+        default_ioclass_id = 0
+        IoclassConfig = namedtuple("IoclassConfig", "id eviction_prio dir_path io_size")
+        io_classes = [
+            IoclassConfig(1, 3, f"{mountpoint}/A", cache_size * 0.25),
+            IoclassConfig(2, 4, f"{mountpoint}/B", cache_size * 0.35),
+            IoclassConfig(3, 5, f"{mountpoint}/C", cache_size * 0.1),
+        ]
+
+        for io_class in io_classes:
+            fs_utils.create_directory(io_class.dir_path, parents=True)
+
+    with TestRun.step("Adding ioclasses for all dirs"):
+        ioclass_config.remove_ioclass_config()
+        ioclass_config.create_ioclass_config(True)
+        for io_class in io_classes:
+            add_io_class(io_class.id, io_class.eviction_prio, f"directory:{io_class.dir_path}&done")
+
+        casadm.load_io_classes(cache_id=cache.cache_id, file=ioclass_config_path)
+
+        # Since default ioclass is already present in cache and no directory should be
+        # created, it is added to ioclasses list after setup is done
+        io_classes.append(IoclassConfig(default_ioclass_id, 22, f"{mountpoint}", cache_size * 0.2))
+
+    with TestRun.step("Verify stats of newly started cache device"):
+        sync()
+        drop_caches(DropCachesMode.ALL)
+        verify_ioclass_usage_stats(cache, [i.id for i in io_classes])
+
+    with TestRun.step("Trigger IO to each partition and verify stats"):
+        for io_class in io_classes:
+            run_io_dir(io_class.dir_path, int((io_class.io_size) / Unit.Blocks4096))
+
+        verify_ioclass_usage_stats(cache, [i.id for i in io_classes])
+
+    with TestRun.step("Purge cache and verify stats"):
+        cache.purge_cache()
+
+        verify_ioclass_usage_stats(cache, [i.id for i in io_classes])
+
+    with TestRun.step("Trigger IO to each partition for the second time and verify stats"):
+        for io_class in io_classes:
+            run_io_dir(io_class.dir_path, int((io_class.io_size) / Unit.Blocks4096))
+
+        verify_ioclass_usage_stats(cache, [i.id for i in io_classes])
+
+
+def get_io_class_usage(cache, io_class_id):
+    return cache.get_io_class_statistics(io_class_id=io_class_id).usage_stats
```

Comment:
Isn't it overkill? You just shortened the function call.

Reply:
I wanted to make it shorter than whole line width.